### ISSUE TITLE
Improvement of filters in the market

### ISSUE THREAD

**Issue by UncleBob800** [45026831/hh-plus-plus#56]
Hello. Until recently I used Tom208's HH++. But due to the fact that it has not fixed a number of problems for a decently long time, I decided to switch to your version of this mod and was very pleasantly surprised by the much improved interface (seriously, no kidding, the booster display panel is simply divine, I have not seen anything more beautiful this, now I don’t have to constantly look into the market to check the Sandalwood, YAY booster).
But I also noticed that several filters in the market are missing.
![image](https://user-images.githubusercontent.com/86453893/155915860-f0bfcbed-c363-479b-9566-a03beb0cb8ed.png)

I also noticed that there is no color highlighting of elements in the filter (it’s problematic for me to remember which element was called, so I was guided by color in the filter). It would be very nice if you could do something similar or put a small element icon in front of its name. This will also be useful for the team change filter.
![image](https://user-images.githubusercontent.com/86453893/155915883-b4cc7e32-58b7-41ae-b61e-90fd0d0796d9.png)

And does not show the current requirement of gems for the selected girl.
https://user-images.githubusercontent.com/86453893/155916054-c2edb6d1-0845-47c8-bbd7-889b0a63558b.mp4

Well, on my own behalf, I would like to ask you to add the categories Pose, Eye Color, Hair Color and Zodiac sign in the filter, so that you can find the girls of the next blessing in the market. Or add a "Next Blessing" filter that would display only girls that match the requirements of the next blessing in the market.
![image](https://user-images.githubusercontent.com/86453893/155916110-ba2fb643-f192-4302-a850-6bd4d8f051eb.png)

But despite these small flaws, I can say with confidence that your mod turned out better than Tom208))) I hope that he will not be offended. It even seems to me that it would not be bad if you both developed this mod together here.

**Comment by 45026831** [comment]
Might I suggest a slightly different means of dealing with the blessings then? The way I manage it is to use zoopokemon's spreadsheet to build a team first, then use the team filter in the market to select just those 7 for upgrades. You aren't going to want to upgrade every single girl that's blessed that week, and you may well have mythics in the team that are top 7 without blessings.

**Comment by UncleBob800** [comment]
Sounds very interesting. However, very often there are times when I try to improve the girls that qualify for the "Next Blessings" during the Girls EXP contest, so that there is more profit than if I improve randomly girls. Previously, I had to go to the Harem section, look for suitable ones through filters (usually legendary ones with 5 stars), remember their names and look for them in the market. You understand that this is not a very convenient process and takes a lot of time. That's why I would like to have "Current Blessings" and "Next Blessings" filters. I think it's possible to combine these two options by adding the "Best Blessings" option with either "Yes"/"No" options. When this option is activated, only the 7 best blessings will be displayed, as you suggested. With this option turned off, it will show all the blessing girls. I think this option will satisfy all needs.

**Comment by 45026831** [comment]
Well again, even if they're blessed, you ought not want to burn XP on girls you aren't going to use. You'd be better off upgrading the girls with the higher likelihood of being in the Top 7 overall (Unblessed stats tab of the spreadsheet).

**Comment by UncleBob800** [comment]
Yes, right. But on the other hand, if there is a situation when there are several girls from these 7 girls, which I can not improve because of, for example, a shortage of gems or their improvement will cost too much, but at the same time I have alternatives no worse than these few girls, It will arise the problem... I just don't see others in the market because the number of displaced blessing girls is limited only than 7 best. I have nothing against this approach, but it seems to me that it will be better if the user can choose whether to show him only 7 best blessing or show all the girls who act a blessing. For example, I have among the best girls now, who have the highest damage, 4 girls and there are still 10 other best girls with the same damage, which can be installed on 5, 6 and 7 slot, respectively. Of these 10, I chose only those that it was beneficial to me to improve on the basis of the available gems. If the list was limited only to 7, then I could not see those girls who were more profitable for me to improve.

**Comment by UncleBob800** [comment]
What about filters in the market? This is all already in "Hentai Heroes++ (OCD) Season version" https://sleazyfork.org/en/scripts/415625-hentai-heroes-ocd-season-version
I was very often needed by "Lvl Max" and "Searched blessed girls" filters. Is it very hard to copy?
Also, I additionally spoke about the color selection of text from the names of the elements in the "Searched Element" filter. It is also present in "Hentai Heroes++ (OCD) Season version"

**Comment by UncleBob800** [comment]
Still really need a "LVL Max" filter in the market, to make it convenient to perform the target "do awakening". 🙏
Still really need a Blessed Girls filter in the markete. 🙏
I first didn't understand what you had in mind when you mentioned about zoopokemon's (sorry, a misunderstanding, I then thought that the Blessed Girls filter was the development of a certain zoopokemon and you offered to add it 😅), but now I saw on the blessings page, that this is a link to Excel table. This is not what we need. When I used "Hentai Heroes++ (OCD) Season version" (it's Tom208 version), then there was a filter of blessed girls in the market. It is very convenient when need to improve girls with maximum benefits. And use for these purposes the table is not very convenient. In HH++ Tom208 I can real-time to improve blessed without any tables.
Your version of HH ++ is definitely better (so better that I can't make myself go back to the version of Tom208, his version of HH++ I am forced to use only when I need LVL Max and Blessed Girls filters), but the absence of these filters, as well as the color allocation in the names of gems and the display of information about the desired number of gems very saddened.
In the first message was not a concept, but a real screenshot of the work of HH ++ Tom208.

**Comment by 45026831** [comment]
I'm fully aware those already exist in Tom's version. Right now, I'm having to prioritise other features over this, I'm afraid. Particularly changes to the new home screen UI. I will get around to improving the market filter, but it will take some time.

**Comment by UncleBob800** [comment]
Understood. Thank for all, I'll be looking forward to these changes. 😉😊

**Comment by POPLEdjdjee** [comment]
Hello, What do you mean by this version has booster display, Tom208's latest version have it too, and IMO it's better. You can add it from here -https://gist.githubusercontent.com/Tom-208/a5c7065866fe1de5032aabbbd1ed9eff/raw/464b50d758ce00f0c8b1833cb567243b691de6f4/HH-game-script.js
